### ISSUE TITLE
Routes have incorrect starting point and reporting 2x actual mileage.

### ISSUE THREAD

**Issue by Memberoffoxhound** [commaai/connect#548]
- Comma 3 0.9.7-release Dongle 49734429dda9bbe7
- 2x mileage issue began approximately 7/14/24
- Secondary issue of incorrect starting location began approximately 7/16/24
- Both issues continue today on every route
- Troubleshooting has included rebooting device and reinstalling openpilot. I have not reflashed or tried to unpair/repair in connect. 

Longer description:
On routes from 7/14/24 connect seems to have around double the miles reported for each route. Even stranger starting on 7/16/24, all routes have a starting location in Chesterfield,MO regardless of my actual location. Chesterfield is where I work and 40 miles from home. 

I am on stock  release and did attempt to reinstall openpilot. I haven’t done anything too extreme yet such as reflashing my comma 3 and I haven’t seen any mention of this anywhere else. The route below should be approximately 12 miles but shows 24. I started in Godfrey, IL despite it showing a starting point 40+ miles away in Chesterfield. 

https://connect.comma.ai/49734429dda9bbe7/00000000--3324219164

![image](https://github.com/user-attachments/assets/ad03e461-c209-44f9-bb67-e8a437e4b237)

**Comment by Memberoffoxhound** [comment]
Small update. Decided to re-flash the device tonight using the web utility and switch from stock to a fork. Both issues persist using sunnypilot. Moving back to stock release at this time. 

TL;DR
- Re-flashed comma3 from web utility
- Switched from stock to sunnypilot fork
- Both starting point and 2x mileage issues remain
- Switching back to stock

**Comment by Memberoffoxhound** [comment]
Re-paired device on connect. No change.

**Comment by m01001100** [comment]
I have the double mileage issue with two separate vehicles using C3 and C3X. Using free version of connect with stock release. Haven't made any attempts to troubleshoot.

**Comment by Rocketboy235** [comment]
Also experiencing this issue on stock OP after factory reset. Also happened to me while on sunnypilot previously so I can at least confirm that it's a connect issue which may have been obvious from the start. 

74b008694de53cc1/0000000b--10878abe6f/0
74b008694de53cc1/0000000c--c967ef9fda/0

Also, been seeing this from previous drives where the starting point starts in the ocean :) 
![chrome_IRR1aIM8VL](https://github.com/user-attachments/assets/3b1ef6a8-989e-4247-a84c-aac803c8e665)

**Comment by EasternPA** [comment]
I bought a used Comma 3 on July 17th. It was removed from the previous owner's account before we met to make the deal. Every single drive I've ever taken with it has started in the parking lot where we met for the sale. I've run OP devel, SGOP20v2, and SP dev. Always the same.

Edit: dongle id: 34346a8b72e53406

**Comment by Memberoffoxhound** [comment]
> I have the double mileage issue with two separate vehicles using C3 and C3X. Using free version of connect with stock release. Haven't made any attempts to troubleshoot. 

Please post your dongle ID at least so comma will have something to look at.

**Comment by Memberoffoxhound** [comment]
> I bought a used Comma 3 on July 17th. It was removed from the previous owner's account before we met to make the deal. Every single drive I've ever taken with it has started in the parking lot where we met for the sale. I've run OP devel, SGOP20v2, and SP dev. Always the same.

Please post your dongle ID at least so comma will have something to look at.

**Comment by Rocketboy235** [comment]
Also got double miles for 74b008694de53cc1/0000001d--6cc2eab86a/0
Technically, it's most if not all routes that are being reported as double mileage

**Comment by m01001100** [comment]
> > I have the double mileage issue with two separate vehicles using C3 and C3X. Using free version of connect with stock release. Haven't made any attempts to troubleshoot.
> 
> Please post your dongle ID at least so comma will have something to look at.

Below are my dongle IDs: 

8d23badbf0a9ea4b - C3
434d76b7c51afa8a - C3X

Just noticed I'm also experiencing the arbitrary start point.

**Comment by perezkris** [comment]
I am having similar issues where my Comma 3 always starts incorrectly in Galax, yet my mileage seems correct... 
My device id 9806e7635acbfc4e
Is there anything we can do?

**Comment by adeebshihadeh** [comment]
@fredyshox I think this was broken by https://github.com/commaai/xx/pull/3600/files. You need to check `hasFix`. It only affects ublox since the qcom module only reports position when it has a fix.

**Comment by fredyshox** [comment]
Alright, will create PR for that

**Comment by jkbkster** [comment]
Same issue with 0d965213ab7ceffa

**Comment by sidgestion** [comment]
same issue with 65ef8b49f9b0dd24. Pretty close to double

**Comment by sshane** [comment]
This is fixed

**Comment by sidgestion** [comment]
Thank you
